core/field.ts

```diff
@@ -1280,12 +1282,19 @@ export abstract class Field<T = any>
    * Get the variable model if this field refers to one.
    *
    * @returns The field's variable, or null if none exists.
-   * @internal
    */
   getVariable(): VariableModel | null {
     return null;
   }
 
+  /**
+   * Sets or clears the variable model.
+   *
+   * @param variable The variable to set or null to clear.
+   */
+  setVariable(_variable: VariableModel | null) {
```

Comment:
Right, we seem to have something more like this at the moment

```
            Field
         /         \
        /           \
FieldDropDown     FieldWithVariable*    // i.e. Would have been called "FieldVariable" if not already taken
       \            /
        \          /
        FieldVariable*                  // i.e. Would have been called "FieldVariableDropdown"  if not already named
```

So maybe it could be more like this?

```
Field
  |
  |
  |
FieldDropDown   FieldVariableMixin
  |            / 
  |           /
  |          /
  |         /
FieldVariable
```

There is probably a name for this, but I don't have a comp-sci background :smile: 

That way, perhaps more than `getVariable` and `setVariable` can be factored out.

Reply:
Haha yeah there's definitely a design flaw in our fields where we tightly coupled the data being modeled to the visual representation. But that is a larger problem :P

I think for now just factoring out `getVariables`, `setVariables`, and `refreshVariableName` into an interface is fine. I think all the other variable-model specific methods in `FieldVariable` are private!

Reply:
Thanks for taking the time on this review, I appreciate it.

I will do as you suggested and try to get back to you this week.